packages/type-utils/tests/TypeOrValueSpecifier.test.ts

```diff
@@ -0,0 +1,306 @@
+import { parseForESLint } from '@typescript-eslint/parser';
+import type { TSESTree } from '@typescript-eslint/utils';
+import Ajv from 'ajv';
```

Comment:
[Praise] Heh, I probably wouldn't have though to test the parsing. But I like it - it's a complicated schema after all!

Reply:
Thanks, yeah, it seemed to me to be complex enough (and given that it's going to be reused, important enough...) to test...